src/DynamicData/Cache/ObservableCache.cs

```diff
@@ -240,4 +322,103 @@ private void InvokePreview(ChangeSet<TObject, TKey> changes)
             }
         }
     }
+
+    private void InvokeCountNext()
+    {
+        lock (_locker)
+        {
+            if (_countChanged.IsValueCreated)
+            {
+                _countChanged.Value.OnNext(_readerWriter.Count);
+            }
+        }
+    }
+
+    private void ResumeCount()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Count without Suspend Count instance");
+            _suspensionTracker.Value.ResumeCount();
+        }
+    }
+
+    private void ResumeNotifications()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Notifications without Suspend Count instance");
+            _suspensionTracker.Value.ResumeNotifications();
+        }
+    }
+
+    private class SuspensionTracker(Action<ChangeSet<TObject, TKey>> onResumeNotifications, Action onResumeCount) : IDisposable
+    {
+        private readonly BehaviorSubject<bool> _notifySuspendSubject = new(false);
```

Comment:
Poor name for this field, IMO. This is just the flag indicating whether notifications are suspended? Except it's a `BehaviorSubject<T>` so it can be subscribed to? It should be `_areNotificationsSuspended` then. Better yet, it could be combined with the actual count variable, `_notifySuspendCount` and then you've reduced the size of the whole `SuspensionTracker` object.

Reply:
Naming is hard.  I like your suggestions though.

I originally had it as `Subject<int>`.  The reason I went with a bool Subject and an int instead of just an int subject is because then to get to the bool, you have to have a more observables in subscription chain (Filter, Select, etc.), which I thought would make it more heavy weight with their closures and allocations.

Reply:
I also didn't want to fire the Observable every time the count changed.  I just need to fire when it goes from to 0 to 1 or 1 to 0.  I thought this was a slight optimization, instead of firing and then filtering.

I'm not too concerned the size of the object because it's own initiated if it is used.  Having the counter outside the Subject is 4 bytes.

Reply:
"Size of the object" is a VERY minor concern, to be sure. Source collections are not constructed often, they're expected to have a relatively-long lifetime, compared to the items and changes they orchestrate.

Since this flag stream is subscribed to by all calls to `.Connect()` and `.Watch()`, you're probably right that it's better to have fewer notifications being published.